### PR TITLE
change(web): drop correction batching 🕐 

### PR DESCRIPTION
**Pull request by jahorton**
While working on #3580, I was noticing some odd behavior after later changes and discovered that the assumption underlying the correction-batching behavior is invalid.  It would also complicate other aspects of #3580, so the simplest answer is to remove it.

@keymanapp-test-bot skip

At the moment, I'm thinking this will be handled in a test covering this and later PRs all at once.

### CODE REVIEW

File: common/web/lm-worker/src/main/correction/distance-modeler.ts [hunk 1]
```diff
@@ -825,20 +784,24 @@ export class SearchSpace {
...
-        batch = batcher.checkAndAdd(newResult.finalNode);
-      }
 
-      if(batch) {
-        yield batch;
+        const entry = newResult.finalNode;
+
+        // As we can't guarantee a monotonically-increasing cost during the search -
+        // due to effects from keystrokes with deleteLeft > 0 - it's technically
+        // possible to find a lower-cost path later in such cases.
```
Comment (jahorton) [inline common/web/lm-worker/src/main/correction/distance-modeler.ts:792]: This was actually observed with the setup established by existing unit tests:

https://github.com/keymanapp/keyman/blob/444ae921972f80b3d81bab3a75ffd8c186167f36/common/web/lm-worker/src/test/mocha/cases/worker-model-compositor.js#L798-L807

Trying to match something prefixed with "hi" is much lower-cost after the 'e' is deleted and the 'i' added in a single step; the conceptual "edge" from "he" to "hi" is thus negative-cost.

The assumption underlying the batching behavior was that path cost would always be monotonically-increasing... something that negative-cost edges invalidate.

File: common/web/lm-worker/src/main/model-compositor.ts [hunk 2]
```diff
@@ -240,71 +240,77 @@ export default class ModelCompositor {
...
-      for await(let matches of searchSpace.getBestMatches(SEARCH_TIMEOUT)) {
+      for await(let match of searchSpace.getBestMatches(SEARCH_TIMEOUT)) {
         // Corrections obtained:  now to predict from them!
-        let predictionRoots = matches.map(function(match) {
-          let correction = match.matchString;
-
-          // Worth considering:  extend Traversal to allow direct prediction lookups?
-          // let traversal = match.finalTraversal;
+        let correction = match.matchString;
+        correctionPredictionMap[match.matchString]
```
Comment (mcdurdin) [inline common/web/lm-worker/src/main/model-compositor.ts:249]: What's this doing?

File: common/web/lm-worker/src/main/model-compositor.ts [hunk 3]
```diff
@@ -240,71 +240,77 @@ export default class ModelCompositor {
...
       const SEARCH_TIMEOUT = this.testMode ? 0 : correction.SearchSpace.DEFAULT_ALLOTTED_CORRECTION_TIME_INTERVAL;
-      for await(let matches of searchSpace.getBestMatches(SEARCH_TIMEOUT)) {
+      for await(let match of searchSpace.getBestMatches(SEARCH_TIMEOUT)) {
         // Corrections obtained:  now to predict from them!
-        let predictionRoots = matches.map(function(match) {
-          let correction = match.matchString;
-
-          // Worth considering:  extend Traversal to allow direct prediction lookups?
-          // let traversal = match.finalTraversal;
+        let correction = match.matchString;
```
Comment (mcdurdin) [inline common/web/lm-worker/src/main/model-compositor.ts:248]: Can we use `const` for variables that never change? I don't think `correction` is changing?

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.59-alpha](https://keyman.com/downloads/releases/alpha/18.0.59)

**Pull request merged**